### ISSUE TITLE
Loss of Z-coordinate in datum conversions

### ISSUE THREAD

**Issue by mheiskan** [OSGeo/PROJ#1833]
This is apparently a similar problem to https://github.com/OSGeo/PROJ/issues/1516
Some legacy code of ours converts coordinates from one projection to another with latlon coordinates as in intermediate step. This has relied on the Z-coordinate being nonzero as in the following:

`> echo 25 60 | cs2cs -f "%.8f" +from +proj=longlat +ellps=WGS84 +towgs84=0,0,0,0,0,0,0 +no_defs +to +proj=stere +lat_0=90 +lat_ts=60 +lon_0=20 +x_0=0 +y_0=0 +R=6371229 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs
279262.99400244 -3191990.62767363 -9096.77560290
`
In PROJ.6 the Z-coordinate is kept at value zero:
`> echo 25 60 | /usr/proj62/bin/cs2cs -f "%.8f" +proj=longlat +ellps=WGS84 +towgs84=0,0,0,0,0,0,0 +no_defs +to +proj=stere +lat_0=90 +lat_ts=60 +lon_0=20 +x_0=0 +y_0=0 +R=6371229 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs
279262.99400244 -3191990.62767364 0.00000000`

Is there any way to get the legacy behavior back via some option? Rewriting all the legacy code to use direct conversions instead may involve a very large number of changes.

**Comment by kbevers** [comment]
> Is there any way to get the legacy behavior back via some option?

Not really. Your input coordinate doesn't have a z-value so it seems odd to me that you even care about the output z.

**Comment by mheiskan** [comment]
Sorry, I thought it was obvious the default is zero. `echo 25 60 0 | ...` gives the same result.

**Comment by rouault** [comment]
The PROJ 6 behaviour is largely driven by the fact that the equivalent of +towgs84 operations in the EPSG database are methods that are intended to be 2D only, that is the change of Z value isn't considered to be meaningful (which makes sense for a number of old system -> new system transformations where ellipsoidal heights didn't make any sense given the knowledge of the old system). That said, I've noticed some confusion regarding that. For example the 7-parameter GDA94->GDA2020 transformation as described in the GDA2020 technical manual page 37 ( https://www.icsm.gov.au/sites/default/files/GDA2020TechnicalManualV1.1.1.pdf ) seems to be geog3D, but is regitered as geog2D only in EPSG (and EPSG contains only geog2D 7-parameter transformations). I've just notified the GDA2020 people about this discrepancy.
